bin/copy-labels.js

```diff
@@ -0,0 +1,322 @@
+    })
+  );
+};
+
+const getExistingLabels = async handler => {
+  let existingLabels;
+  try {
+    existingLabels = await handler.get();
+  } catch (err) {
+    helpExit('Cannot get existing labels', err);
+  }
+  return existingLabels;
+};
+
+const dryCopyLabels = async (args, handler, labels, usedLabels) => {
+  if (args.dstFile) {
+    console.log(
+      `Write following labels ` +
+        `from repository '${args.srcRepo}' to file '${args.dstFile}'`
+    );
+    printLabels(labels);
+  } else {
+    const existingLabels = await getExistingLabels(handler);
+    let msg = 'Copy the following labels from';
+    if (args.srcFile) {
+      msg += ` file '${args.srcFile}'`;
+    } else {
+      msg += ` repository '${args.srcRepo}'`;
+    }
+    msg += ` to repository '${args.dstRepo}'`;
+    if (args.update) msg += ' updating existing labels';
+    console.log(msg);
+    printAddedLabels(args, labels, existingLabels, usedLabels);
+    if (args.update) {
+      printUpdatedLabels(args, labels, existingLabels, usedLabels);
+    }
+    printUnchangedLabels(args, labels, existingLabels, usedLabels);
+  }
+};
+
+const copyLabels = async (args, handler, labels) => {
+  if (args.dstFile) {
+    try {
+      await writeDump(args.dstFile, labels);
+    } catch (err) {
+      helpExit('Cannot write dump', err);
+    }
+  } else {
+    try {
+      labels = await handler.copyFrom(labels, args.update);
+    } catch (err) {
+      helpExit('Cannot copy labels', err);
+    }
+  }
+  return labels;
+};
+
+const getLabels = async (args, handler) => {
+  let labels;
+  if (args.srcFile) {
+    try {
+      labels = await readDump(args.srcFile);
+    } catch (err) {
+      helpExit('Cannot read dump', err);
+    }
+  } else {
+    try {
+      labels = await handler.getFrom(args.srcRepo);
+    } catch (err) {
+      helpExit('Cannot read dump', err);
+    }
+  }
+  return labels;
+};
+
+const getConfig = async args => {
+  if (args.config) {
+    const { user, token } = await readDump(args.config);
+    args = { user, token, ...args };
+  }
+
+  const [src, dst] = args.args;
+  if (!src) helpExit('missing source repository or file');
+  if (!dst) helpExit('missing destination repository or file');
+
+  if (path.extname(src) === '.json') args.srcFile = src;
+  else args.srcRepo = src;
+  if (path.extname(dst) === '.json') args.dstFile = dst;
+  else args.dstRepo = dst;
+
+  if (args.srcFile && args.dstFile) {
+    helpExit('Either source or destination should be repository');
+  }
+  return args;
+};
+
+const copy = async args => {
+  args = await getConfig(args);
+  const handler = new LabelHandler({
+    repo: args.dstRepo,
+    user: args.user,
+    token: args.token,
+  });
+
+  if (args.dry) console.log('This is a dry run, no changes will be made\n');
+
+  let usedLabels = new Map();
+  if (args.dstRepo && args.delete) {
+    usedLabels = await deleteLabels(args, handler);
+  }
+
+  const labels = await getLabels(args, handler);
+  if (args.dry) {
```

Comment:
Nit: I don't like that the handling of `args.dry` is spread through so much code - here above and below, in the `deleteLabels`. I'd prefer to localize it somewhere considering there is not much code reuse there.